### PR TITLE
[FLINK-14509][python] Improve the README.md in flink-python to prepare for PyPI release.

### PR DESCRIPTION
**Pull request by WeiZhong94**
## What is the purpose of the change

*This pull request improves the text of README.md in flink-python to prepare for PyPI release.*

## Brief change log

  - *Improve the text of README.md in flink-python*
  - *Recommend users using `pip install apache-flink` to install PyFlink in the Python API tutorial.*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-python/README.md [hunk 1]
```diff
@@ -1,37 +1,28 @@
...
-PyFlink depends on Py4J (currently version 0.10.8.1) and CloudPickle (currently version 1.2.2).
+Apache Flink Python API depends on Py4J (currently version 0.10.8.1), CloudPickle (currently version 1.2.2) and Apache Beam (currently version 2.15.0).
+
+## Development Notice
 
-## Development notices
+### Protobuf Code Generation
 
-Protocol buffer is used in this module and file `flink_fn_execution_pb2.py` is generated from `flink-fn-execution.proto`. Whenever `flink-fn-execution.proto` is updated, please re-generate `flink_fn_execution_pb2.py` by executing
+Protocol buffer is used in file `flink_fn_execution_pb2.py` is generated from `flink-fn-execution.proto`. Whenever `flink-fn-execution.proto` is updated, please re-generate `flink_fn_execution_pb2.py` by executing.
```
Comment (hequn8128) [inline flink-python/README.md]: - and it is generated from.
- change . to :

File: flink-python/README.md [hunk 2]
```diff
@@ -1,37 +1,28 @@
...
-./dev/lint-python.sh
-```
+The auto-generated Python docs can be found at [https://ci.apache.org/projects/flink/flink-docs-stable/api/python/](https://ci.apache.org/projects/flink/flink-docs-stable/api/python/)
 
 ## Python Requirements
 
-PyFlink depends on Py4J (currently version 0.10.8.1) and CloudPickle (currently version 1.2.2).
+Apache Flink Python API depends on Py4J (currently version 0.10.8.1), CloudPickle (currently version 1.2.2) and Apache Beam (currently version 2.15.0).
+
+## Development Notice
```
Comment (hequn8128) [inline flink-python/README.md]: Development Notices

File: flink-python/README.md [hunk 3]
```diff
@@ -1,37 +1,28 @@
...
-
-```
-./dev/lint-python.sh
-```
+The auto-generated Python docs can be found at [https://ci.apache.org/projects/flink/flink-docs-stable/api/python/](https://ci.apache.org/projects/flink/flink-docs-stable/api/python/)
 
 ## Python Requirements
 
-PyFlink depends on Py4J (currently version 0.10.8.1) and CloudPickle (currently version 1.2.2).
+Apache Flink Python API depends on Py4J (currently version 0.10.8.1), CloudPickle (currently version 1.2.2) and Apache Beam (currently version 2.15.0).
```
Comment (hequn8128) [inline flink-python/README.md]: Also depends on `python-dateutil==2.8.0`?

File: flink-python/README.md [hunk 4]
```diff
@@ -1,37 +1,28 @@
-# Apache Flink Python API
+# Apache Flink
```
[the remark this answers is not in the document]
Comment (WeiZhong94) [reply flink-python/README.md]: This readme will be compile to the description page of Apache Flink in pypi.org, so I think we'd better introduce our project here.

**Comment by WeiZhong94** [comment]
@hequn8128 Thanks for your review! I have addressed your comments in the latest commit.

File: flink-python/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (hequn8128) [reply flink-python/README.md]: ok, make sense.

**Pull request closed without merging**